src/protocol/LSP.References.pas

```diff
@@ -132,7 +132,7 @@ procedure TReferencesRequest.FindReferences(Filename, MainFilename: String; X, Y
     X,Y,
     DeclCode,DeclX,DeclY,DeclTopLine) then
   begin
-    PublishDiagnostic(Transport,'FindMainDeclaration failed in '+StartSrcCode.FileName+' at '+IntToStr(Y)+':'+IntToStr(X));
+    DiagnosticsHandler.PublishDiagnostic(Transport,'FindMainDeclaration failed in '+StartSrcCode.FileName+' at '+IntToStr(Y)+':'+IntToStr(X));
```

Comment:
Ok this is different from normal syntax errors as they are triggered by a user action and aren't actually syntax errors themselves. Before code tools errors were being shown like this but I don't think it's technically a correct use of diagnostics.

The problem is the error will go away when the user updates the file right? Maybe that's ok?

Another option worth considering is making these errors a `TNotificationMessage` so they are popped up as a window. This seems more correct from a UI/UX point of view but they could get annoying, especially depending on how the IDE presents them (VSCode is ok IMO).

Reply:
Ryan,
I don't understand why you make this remark. I didn't change the behavior. 
This use of diagnostics was already there, I certainly didn't introduce it; 
I only created an object to handle the diagnostics.

The message window popup idea is very annoying as you indicated (IMO even in VSCode). 
I disabled the use of that message window - it is a setting.

I would introduce a new setting 'report codetools errors as' 
with three settings:
- diagnostic messages
- popup windows
- debug output.
But that should be part of another rework.